portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,28 +14,108 @@
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.SortDirection;
 import com.google.gson.Gson;
+import com.google.sps.data.Comment;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.util.ArrayList; 
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/data")
+/** Servlet responsible for handling comments data */
+@WebServlet("/comment")
 public class DataServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    ArrayList<String> messages = new ArrayList<String>();
-    messages.add("Hello");
-    messages.add("Testing JSON");
-    messages.add("I hope this works!");
+    String sort = request.getParameter("sort");
+
+    // Sort comments according to request parameter
+    Query query;
+    FilterPredicate filter = new FilterPredicate("img", Query.FilterOperator.EQUAL, request.getParameter("img"));
+    if (sort.equals("new")) {
+      query = new Query("Comment").setFilter(filter).addSort("timestamp", SortDirection.DESCENDING);
+    } else {
+      query = new Query("Comment").setFilter(filter).addSort("timestamp", SortDirection.ASCENDING);
+    }
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    int numComments = results.countEntities(FetchOptions.Builder.withDefaults());
+
+    // Set the number of comments to print (max)
+    String quantity = request.getParameter("quantity");
+    int max = -1;
+    if (quantity != null && quantity.length() > 0) {
+      max = Integer.parseInt(quantity);
+    } else {
+      max = numComments;
+    }
+
+    // Create Comment objects and add to comments ArrayList
+    ArrayList<Comment> comments = new ArrayList<Comment>();
+    for (Entity entity : results.asIterable(FetchOptions.Builder.withLimit(max))) {
+      long id = entity.getKey().getId();  
+      String comment = (String) entity.getProperty("comment");
+      String name = (String) entity.getProperty("name");
+      long timestamp = (long) entity.getProperty("timestamp");
+      String img = (String) entity.getProperty("img");
+
+      Comment c = new Comment(id, name, comment, timestamp, img);
+      comments.add(c);
+    }
+    
+    // Create json String to print to /comments
     Gson gson = new Gson();
-    String json = gson.toJson(messages);
+    String json = "{";
+    json += "\"comments\": ";
+    json += gson.toJson(comments);
+    json += ", ";
+    json += "\"total\": ";
+    json += numComments;
+    json += "}";
```

Comment:
Are u able to use String.format

Reply:
Yes, I changed it! it looks so much nicer